sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -1473,7 +1471,7 @@ public PCollection<KafkaRecord<K, V>> expand(PBegin input) {
         }
         PCollection<KafkaSourceDescriptor> output;
         if (kafkaRead.isDynamicRead()) {
-          Set<String> topics = new HashSet<>();
+          List<String> topics = new ArrayList<>();
```

Comment:
why is this a list instead?

Reply:
In the original implementation, it was a set that got converted to a list. I just changed it to start as a list, but it could be a set just as easily

Reply:
Well, I just think we'll likely have duplicates if we use a List instead of a Set, because in line 1480 we're listing TopicPartitions and appending Topics to the list, no?

Reply:
true. I'll change it to a set everywhere